wormhole-connect/src/utils/routes/cctpManual.ts

```diff
@@ -0,0 +1,612 @@
+import {
+  ChainId,
+  ChainName,
+  TokenId,
+  EthContext,
+  WormholeContext,
+  TokenMessenger__factory,
+  MessageTransmitter__factory,
+} from '@wormhole-foundation/wormhole-connect-sdk';
+import { CHAINS, TOKENS, TOKENS_ARR } from 'config';
+import { TokenConfig } from 'config/types';
+import { BigNumber, utils, BytesLike, ethers } from 'ethers';
+import axios, { AxiosResponse } from 'axios';
+import {
+  MAX_DECIMALS,
+  getTokenById,
+  getTokenDecimals,
+  toNormalizedDecimals,
+} from 'utils';
+import { isMainnet } from 'config';
+import {
+  ParsedMessage,
+  ParsedRelayerMessage,
+  PayloadType,
+  toChainId,
+  wh,
+} from 'utils/sdk';
+import { calculateGas } from 'utils/gas';
+import { TransferWallet, signAndSendTransaction } from 'utils/wallet';
+import { TransferDisplayData } from './types';
+import { BaseRoute } from './baseRoute';
+import { toDecimals, toFixedDecimals } from '../balance';
+import { TransferInfoBaseParams, CCTPInfo } from './routeAbstract';
+import { getGasFeeFallback } from '../gasEstimates';
+import { Route } from 'store/transferInput';
+export interface CCTPManualPreviewParams {
+  destToken: TokenConfig;
+  sourceGasToken: string;
+  destinationGasToken: string;
+  receiveAmount: number;
+  sendingGasEst: string;
+  destGasEst: string;
+}
+
+interface TransferDestInfoParams {
+  txData: ParsedMessage | ParsedRelayerMessage;
+  receiveTx?: string;
+}
+
+async function sleep(timeout: number) {
+  return new Promise((resolve) => setTimeout(resolve, timeout));
+}
+
+const CIRCLE_ATTESTATION = isMainnet
+  ? 'https://iris-api.circle.com/attestations/'
+  : 'https://iris-api-sandbox.circle.com/attestations/';
+
+async function getCircleAttestation(messageHash: BytesLike) {
+  while (true) {
+    // get the post
+    const response = await tryGetCircleAttestation(messageHash);
+
+    if (response !== null) {
+      return response;
+    }
+
+    await sleep(6500);
+  }
+}
+
+async function tryGetCircleAttestation(
+  messageHash: BytesLike,
+): Promise<string | null> {
+  return await axios
+    .get(`${CIRCLE_ATTESTATION}${messageHash}`)
+    .catch((reason) => {
+      return null;
+    })
+    .then(async (response: AxiosResponse | null) => {
+      if (
+        response !== null &&
+        response.status === 200 &&
+        response.data.status === 'complete'
+      ) {
+        return response.data.attestation as string;
+      }
+
+      return null;
+    });
+}
+
+function getChainNameCCTP(domain: number): ChainName {
+  switch (domain) {
+    case 0:
+      return isMainnet ? 'ethereum' : 'goerli';
+    case 1:
+      return isMainnet ? 'avalanche' : 'fuji';
+  }
+  throw Error('Invalid CCTP domain');
+}
+function getDomainCCTP(chain: ChainName | ChainId): number {
+  const chainId = wh.toChainId(chain);
+  switch (chainId) {
+    case 2:
+      return 0;
+    case 6:
```

Comment:
Not a big deal now since there's only a few chains supported, but maintaining these switch cases with growing chains can be a bit of a headache.  Consider adding a CCTP domain on the chain config?